app/utils/img_processing/tesseract_api.py

```diff
@@ -3,76 +3,164 @@
 from app.utils.moderation.text_moderation import BadWordTextModerator
 
 from cv2 import imdecode, cvtColor, IMREAD_COLOR, COLOR_BGR2RGB
-from pytesseract import image_to_string
+from pytesseract import image_to_string, image_to_data, pytesseract
 import numpy as np
 
-# Get the name of the directory where this file
-# is located.
+# Get name of directory where this file is located
 DIR = dirname(__file__)
 
 # Setup globals
 BAD_WORDS_CSV = DIR + '/../moderation/bad_single.csv'
 MODEL_DIR = DIR + '/../../../models/'
 TESSDATA_DIR = MODEL_DIR
 
-# Tell tesseract to look in the TESSDATA_DIR dir for
-# lang trainedata files.
+# Tell tesseract to look in TESSDATA_DIR for lang trainedata files
 TESS_CONFIG = f'--tessdata-dir "{TESSDATA_DIR}"'
 
+# Windows users: add local Tesseract path, example below
+# This should not be necessary if running through Docker (or for Mac users)
+# pytesseract.tesseract_cmd = 'C:/Program Files/Tesseract-OCR/tesseract.exe'
+
 
 class TesseractAPI:
-    """# An Interface to the Tesseract OCR Engine"""
+    """
+    Interface to Tesseract OCR engine
+    Takes single image page and returns transcribed text
+    """
 
     def __init__(self, lang='storysquad'):
-        """function that prepares the TesseractAPI to handle requests from the
-        endpoints.
-
         """
-        load_dotenv()
+        Arguments:
+        Tesseract model (default is 'storysquad')
+        
+        Actions:
+        Prepares TesseractAPI to handle requests from the endpoints
+        """
 
+        load_dotenv() 
         self.text_moderator = BadWordTextModerator(BAD_WORDS_CSV)
         self.lang = lang
 
-    def img_preprocess(self, img):
-        ret = []
+    def img_preprocess(self, image):
+        """
+        Arguments:
+        Single full-page image
 
-        img = cvtColor(img, COLOR_BGR2RGB)
+        Actions:
+        1. Convert image data bytestring to image object
+        2. Convert order of image colors from BGR to RGB
 
-        ret.append(img)
+        Returns:
+        Full-page processed image
+        """
 
-        # Return processed image
-        return ret
+        # 1. Convert bytestring to image object
+        nparr = np.fromstring(image, np.uint8)
+        image_array = imdecode(nparr, IMREAD_COLOR)
+
+        # 2. Convert image from BGR to RGB
+        processed_img = cvtColor(image_array, COLOR_BGR2RGB)
 
-    async def transcribe(self, document):
-        """Detects document features in images and returns extracted text
-        Input:
-        --------
-        `document`: bytes - The file object to be sent to Tesseract OCR Engine
+        # Return processed image
+        return processed_img
```

Comment:
idk if this is how you guys have been instructed to structure your comments but typically I avoid comments that are easy to infer from the code itself. For example if line 62-65 looked like the following, I think its self explanatory sans a comment

```python
rgb_img = cvtColor(bgr_img_array, COLOR_BGR2RGB)
return rgp_img
```

Mostly the reason I nitpick this is that if someone changes the code later they will have to remember (or the code reviewer notices) stale comments. Tests won't catch stale comments so in a lot of codebases comments become out of date much faster than expected.

Reply:
Thanks @nlittlepoole I've removed the extra comments :)